src/Microsoft.DotNet.Configurer/NuGetCachePrimer.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.EnvironmentAbstractions;
+
+namespace Microsoft.DotNet.Configurer
+{
+    public class NuGetCachePrimer : INuGetCachePrimer
+    {
+        private const string NUGET_SOURCE_PARAMETER = "-s";
+        private readonly ICommandFactory _commandFactory;
+        private readonly IDirectory _directory;
+        private readonly IFile _file;
+        private readonly INuGetPackagesArchiver _nugetPackagesArchiver;
+        private readonly INuGetCacheSentinel _nuGetCacheSentinel;
+
+        public NuGetCachePrimer(
+            ICommandFactory commandFactory,
+            INuGetPackagesArchiver nugetPackagesArchiver,
+            INuGetCacheSentinel nuGetCacheSentinel)
+            : this(commandFactory,
+                nugetPackagesArchiver,
+                nuGetCacheSentinel,
+                FileSystemWrapper.Default.Directory,
+                FileSystemWrapper.Default.File)
+        {
+        }
+
+        internal NuGetCachePrimer(
+            ICommandFactory commandFactory,
+            INuGetPackagesArchiver nugetPackagesArchiver,
+            INuGetCacheSentinel nuGetCacheSentinel,
+            IDirectory directory,
+            IFile file)
+        {
+            _commandFactory = commandFactory;
+            _directory = directory;
+            _nugetPackagesArchiver = nugetPackagesArchiver;
+            _nuGetCacheSentinel = nuGetCacheSentinel;
+            _file = file;
+        }
+
+        public void PrimeCache()
+        {
+            if(SkipPrimingTheCache())
+            {
+                return;
+            }
+
+            var pathToPackagesArchive = _nugetPackagesArchiver.ExtractArchive();
+
+            PrimeCacheUsingArchive(pathToPackagesArchive);
+        }
+
+        private bool SkipPrimingTheCache()
```

Comment:
We need to think about more cases where we would want to skip this, here are the ones that come to mind:
1. Machine-wide install exists with a fallback packages folder.
2. We're in a KUDU/Antares environment where they care a lot about disk space and are doing their own thing (perhaps an env variable).
3. Someone integrating CLI into a CI system and want hardcore control of packages & sources.

Maybe you're already doing something like this higher in the stack, LMK if that's the case.